smart_open/tests/test_smart_open.py

```diff
@@ -200,93 +202,158 @@ def test_http_pass(self):
-
-        # decompress the bzip2 and get the same md5 hash
-        self.assertEqual(smart_open_object.read(), test_string)
-
-    @responses.activate
     def test_http_xz(self):
         """Can open xz via http?"""
-        test_string = b'Hello World Compressed.'
-        #
-        # TODO: why are these tests writing to temporary files?  We can do the
-        # lzma compression in memory.
-        #
-        with tempfile.NamedTemporaryFile('wb', suffix='.xz', delete=False) as infile:
-            test_file = infile.name
+        self._test_compressed_http(".xz", False)
 
-        with smart_open.smart_open(test_file, 'wb') as outfile:
-            outfile.write(test_string)
+    @unittest.skipIf(six.PY2, 'gzip support for Py2 is not implemented yet')
+    def test_http_gz_query(self):
+        """Can open gzip via http with a query appended to URI?"""
+        self._test_compressed_http(".gz", True)
 
-        with open(test_file, 'rb') as infile:
-            compressed_data = infile.read()
+    def test_http_bz2_query(self):
+        """Can open bzip2 via http with a query appended to URI?"""
+        self._test_compressed_http(".bz2", True)
 
-        if os.path.isfile(test_file):
-            os.unlink(test_file)
+    def test_http_xz_query(self):
+        """Can open xz via http with a query appended to URI?"""
+        self._test_compressed_http(".xz", True)
 
-        responses.add(responses.GET, "http://127.0.0.1/data.xz", body=compressed_data, stream=True)
-        smart_open_object = smart_open.smart_open("http://127.0.0.1/data.xz")
 
-        # decompress the xz and get the same md5 hash
-        self.assertEqual(smart_open_object.read(), test_string)
+class SmartOpenFileObjTest(unittest.TestCase):
+    """
+    Test passing raw file objects.
+    """
+    text = 'Hello, world!'
 
+    def patch_string_io(func):
+        if not PY2:
+            # Python 3 has already got us covered
+            return func
+
+        def enter(self):
+            return self
+
+        def exit(self, exc_type, exc_val, exc_tb):
+            self.close()
+
+        @functools.wraps(func)
+        def wrapped_patch_string_io(self):
+            six.StringIO.__enter__ = enter
+            six.StringIO.__exit__ = exit
+            try:
+                return func(self)
+            finally:
+                del six.StringIO.__enter__
+                del six.StringIO.__exit__
+
+        return wrapped_patch_string_io
+
+    @patch_string_io
+    def test_read_bytes(self):
+        """Can we read bytes from a byte stream?"""
+        buffer = six.BytesIO(self.text.encode())
+        with smart_open.smart_open(buffer, 'rb') as sf:
+            data = sf.read()
+        self.assertEqual(data, self.text.encode())
+
+    @patch_string_io
+    def test_write_bytes(self):
+        """Can we write bytes to a byte stream?"""
+        buffer = six.BytesIO()
+        with smart_open.smart_open(buffer, 'wb') as sf:
+            sf.write(self.text.encode())
```

Comment:
We're using TEXT.encode() a lot. It's making our tests more difficult to read than they really need to be.

If it's worth declaring TEXT as a constant, then it's worth declaring its corresponding bytes as a constant too.

```
TEXT = 'hello world'
BYTES = TEXT.encode('utf-8')
```